src/app/services/scheduler/scheduler.service.ts

```diff
@@ -328,13 +328,13 @@ export class SchedulerService {
     const formatted = [];
 
     if (resource && resource.memory !== undefined) {
-      formatted.push(`Memory: ${CommonUtil.formatBytes(resource.memory)}`);
+      formatted.push(`Memory: ${CommonUtil.formatMemoryBytes(resource.memory)}`);
     } else {
       formatted.push(`Memory: ${NOT_AVAILABLE}`);
     }
 
     if (resource && resource.vcore !== undefined) {
-      formatted.push(`CPU: ${CommonUtil.formatCount(resource.vcore)}`);
+      formatted.push(`CPU: ${CommonUtil.formatCpuCore(resource.vcore)}`);
```

Comment:
When making this change, I noticed that if the resource is undefined, the result will be empty. Even though this might not happen in the real world, I prefer to keep the original behavior.

Screenshot with json-server data.
<img width="336" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/f572851b-f0d1-4001-bb17-4d7aaf7165aa">

I refactored the code and added test cases/comments to make this function more readable. I hope this is helpful.
Additionally, I include a sorting function before foreach loop, which guarantees the order of Memory/Cpu/ephemeral-storage. Please let me know if it's not necessarily.

Reply:
> When making this change, I noticed that if the resource is undefined, the result will be empty. Even though this might not happen in the real world, I prefer to keep the original behavior.

I think showing that we have no keys at all in the end is better than showing `memory` and `cpu` always. Showing `memory` and `cpu` is a choice pointing back to a time that `memory` and `cpu` were required types. They have never been that in YuniKorn. If a resource is completely empty we should show that as `n/a` I think. Even shortcut that all the way in the beginning of the conversion code.
This empty resource by the way only affects queues, nodes and applications when it comes to certain resource values. Even for the best effort pods there is no `memory` or `CPU` but we always set the `pods` resource. 

Having a fixed order is good, I like that side of the change. `memory` and `CPU` should be first in the list if set. Extending that with `ephemeral-storage` is good. I would add `pods` in that list too (even before storage)

Reply:
Update this PR with:
1. Return just n/a if there no resources keys in resource object.
2. Add ‘pods’ in resourcesCompareFn (before ‘ephemeral-storage’)

Reply:
Hi @wilfred-s,
There have 2 other metrics are always showing `memory` and `cpu` too.
Both are calling formatPercent() in scheduler.service.ts
1. 'Absolute Used Capacity' in queue 
2. 'Utilized' in node


For example, when 'memory'=7%, 'vcore'='19%', pod='1%', the result only show 'Memory' and 'Cpu' there.

**Example(Queue)**:
<img width="841" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/ca9c8d92-af30-457e-a8d2-b012efd9f540">

<img width="342" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/70e7e16d-c20f-446b-b140-815151231672">


**When no 'memory', 'vcore' in resources:**
<img width="335" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/1b62552d-4ec5-441d-bfde-db330584f126">

Reply:
Update new commit: bf35c02e4206ce0027a00ff6448f2e17673c30ed
I just follow the same logic to handle no keys in resources.  (return 'n/a' only )
Didn't expose more resources becasue I'm not sure if it's better to have,
(If requried I can create another jira to expose it.)

**Before**:
<img width="345" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/0ca27b84-39cb-4798-a0b3-98a14aa39567">


**After**:
<img width="335" alt="image" src="https://github.com/apache/yunikorn-web/assets/26764036/2240ea3b-06e1-4779-b55a-0b586d1cb410">